tensorflow_io/arrow/python/ops/arrow_dataset_ops.py

```diff
@@ -357,19 +364,21 @@ class ArrowStreamDataset(ArrowBaseDataset):
   """An Arrow Dataset for reading record batches from an input stream.
   Currently supported input streams are a socket client or stdin.
   """
+  host_types_supported = ('AF_INET', 'AF_UNIX', 'STDIN')
 
   def __init__(self,
-               host,
+               hosts,
                columns,
                output_types,
                output_shapes=None,
                batch_size=None,
-               batch_mode='keep_remainder'):
+               batch_mode='keep_remainder',
+               host_type='AF_INET'):
```

Comment:
For python I think we want to avoid introduce args if possible. For unix sockets normally they are specified with scheme prefix `unix://<socket_address>`. For inet socket they are specified with `tcp://<host>:<port>`.

For standard input, I vaguely remember it used to be `fd://<number>`? (and pipe used to be `pipe://<address>`)

To make it easier (and compatible) with previous definitions, I think we also could default the scheme as `tcp://` (so that `host:port` will be interpreted as `tcp://host:port`.

For that I think maybe we could have the following API for hosts (or `endpoints` as was mentioned in another comment):
```
endpoints: A `tf.string` tensor, Python list or scalar string defining the  input stream.
    endpoints` could have the following format:,
      - `host:port`: IPv4 address (default)
      - `tcp://host:port`: IPv4 address,
      - `unix://<unix socket address>`: local path as unix socket address,
      - `fd://<number>`: STDIN or file descriptor number.
```
```

Reply:
great idea, it's best to use a standard and not introduce unnecessary args. +1 on the above API, I saw you used `fd://-` for stdin on another PR.

Reply:
I am a little unsure about stdin, as there are many ways to specify it.

In some software, `-` are used in `command line` to specify `stdin`. e.g
- `docker import -`
- `kubectl apply -f -`
- `tshark -r -`.

In some other software, `/dev/stdin` are used as stdin is essentially still a file on Linux. The issue with `file://` is that, in TF `file://` stands for posix file system, and the file specified by `file://` is `openable` and `closable`. But this is not the case for stdin. So we have to use another sheme for stdin than `file://`

I think we could either treat stdin as a generic case of `fd` (so `fd://0`) or a special case of `fd` (`fd://-`). Or support both?

Reply:
>I think we could either treat stdin as a generic case of fd (so fd://0) or a special case of fd (fd://-). Or support both?

It shouldn't be too hard to support both, I'll give it a try